### PR TITLE
Do not crash when unsetting scrollToRow

### PR DESCRIPTION
**Pull request by IanVS**
## Description
This avoids trying to scroll to an `undefined` row.

## Motivation and Context
Since the latest release (0.7.5), I began to see the following error:

```
Uncaught Error: Invariant Violation: Index out of range undefined
```

I found that this happened after setting a `scrollToRow` prop and then returning it to `undefined`.

## How Has This Been Tested?
I tested this in my app, as well as added a test for it.  Note: I needed to be able to `setState`, which required full rendering, so I add jsdom and react-dom to the dev dependencies.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by IanVS** [comment]
Not sure why Travis choked.  Maybe someone can restart it?

File: src/FixedDataTable.js [hunk 1]
```diff
@@ -912,7 +912,7 @@ var FixedDataTable = React.createClass({
     }
 
     var lastScrollToRow  = oldState ? oldState.scrollToRow : undefined;
-    if (props.scrollToRow !== lastScrollToRow) {
+    if (typeof props.scrollToRow !== 'undefined' && props.scrollToRow !== null && props.scrollToRow !== lastScrollToRow) {
```
Comment (KamranAsif) [inline src/FixedDataTable.js]: I think we need to implement something similar for the other: scrollTop, scrollToRow and scrollToColumn

Comment (IanVS) [inline src/FixedDataTable.js]: I thought so too, but they weren't failing similar tests, while this one was, so I opted not to.

Comment (IanVS) [inline src/FixedDataTable.js]: That would be cool, but I guess I would argue that could be a separate PR after this one (if this is accepted).

File: build_helpers/test-globals.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// mocha global object
+propagateToGlobal(win)
+
+// from mocha-jsdom https://github.com/rstacruz/mocha-jsdom/blob/master/index.js#L80
+function propagateToGlobal (window) {
+  for (let key in window) {
+    if (!window.hasOwnProperty(key)) continue
+    if (key in global) continue
+
+    global[key] = window[key]
```
Comment (KamranAsif) [inline build_helpers/test-globals.js]: nit: semicolon

File: build_helpers/test-globals.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+var jsdom = require('jsdom')
```
Comment (KamranAsif) [inline build_helpers/test-globals.js]: nit: missing semicolons throughout this file

Comment (IanVS) [inline build_helpers/test-globals.js]: That's what I get for copying it from somewhere else.  Sure would be nice to have ESLint set up for this project.  Any interest in a PR doing so?

Comment (KamranAsif) [inline build_helpers/test-globals.js]: Yeah, would love it!

**Comment by IanVS** [comment]
What will it take to get this merged?  Add semicolons?  Would you like checks around `scrollTop`, `scrollLeft`, etc?

**Comment by KamranAsif** [comment]
@IanVS Just want to checkout the branch and test a couple edge cases. Are you in a hurry to get this merged in?

**Comment by IanVS** [comment]
Nope, no particular hurry, just want to make sure I understand the expectations.  Wasn't sure if you were done reviewing.

**Comment by KamranAsif** [comment]
Added some extra tests; found scrollTop was also failing.
I refactored the tests to reuse your instance and added some improvements to make it more generic

@wcjordan Can you give my changes a look?

File: src/FixedDataTableRoot-test.js [hunk 4]
```diff
@@ -33,16 +33,32 @@ describe('FixedDataTableRoot', function() {
...
+      this.state = {...props}
+    }
+
+    /**
+     * Returns state from FDT.Table object
+     *
+     * @return {!Object}
+     */
+    getTableState() {
+      return this.refs['table'].state;
```
Comment (wcjordan) [inline src/FixedDataTableRoot-test.js]: what's the motivation for this state access indirection?  might be worth adding a comment.

**Comment by wcjordan** [comment]
I added a few questions, but lgtm

File: src/FixedDataTable.js [hunk 5]
```diff
@@ -920,7 +920,7 @@ var FixedDataTable = React.createClass({
     }
 
     var lastScrollTop = oldState ? oldState.scrollTop : undefined;
-    if (props.scrollTop !== lastScrollTop) {
+    if (typeof props.scrollTop !== 'undefined' && props.scrollTop !== null && props.scrollTop !== lastScrollTop) {
```
Comment (KamranAsif) [inline src/FixedDataTable.js]: Not sure, I just copied it. I think we can just do props.scrollTop != null since undefined == null
Will update this and above

Comment (IanVS) [inline src/FixedDataTable.js]: It's a habit I suppose, but it's usually considered safer.  In strict mode accessing a variable that is undefined will throw a referenceError.  That may not apply in this case, but as a general rule to be safe that's how I like to check for undefined variables.

File: src/FixedDataTableRoot-test.js (hunk 4 above)
Comment (KamranAsif) [inline src/FixedDataTableRoot-test.js]: We use it to inspect the inner state for the tests. I'll update the annotation to further clarify

File: src/FixedDataTable.js (hunk 5 above)
Comment (wcjordan) [inline src/FixedDataTable.js]: Hmm that isn't my finding here: http://jsfiddle.net/s14sav6k/4/
But I'm not too familiar with strict mode.  Do you have documentation which describes what can cause the ReferenceErrors?  Could it be a browser specific implementation?

Comment (wcjordan) [inline src/FixedDataTable.js]: Cool, I think the case you're referring to is specific to references on the execution context (props for this case), rather than accessed through an object with dot notation scrollTop.

If there's a risk of props not existing, we'd need to also add a `typeof props !== undefined &&` check to avoid an error from the dot access.  If not, I think we're safe with the short hand Kam suggests in both strict and non-strict modes.

Comment (IanVS) [inline src/FixedDataTable.js]: As I said, just a habit.  ;)

**Comment by IanVS** [comment]
Thanks everyone.

**Comment by KamranAsif** [comment]
Published in 0.7.6
